zrml/market-commons/src/migrations.rs

```diff
@@ -14,3 +14,61 @@
 //
 // You should have received a copy of the GNU General Public License
 // along with Zeitgeist. If not, see <https://www.gnu.org/licenses/>.
+
+// We use these utilities to prevent having to make the swaps pallet a dependency of
+// prediciton-markets. The calls are based on the implementation of `StorageVersion`, found here:
+// https://github.com/paritytech/substrate/blob/bc7a1e6c19aec92bfa247d8ca68ec63e07061032/frame/support/src/traits/metadata.rs#L168-L230
+// and previous migrations.
+mod utility {
+    use alloc::vec::Vec;
+    use frame_support::{
+        storage::{storage_prefix, unhashed},
+        traits::StorageVersion,
+        StorageHasher,
+    };
+    use parity_scale_codec::Encode;
+
+    #[allow(unused)]
+    pub fn storage_prefix_of_market_common_pallet() -> [u8; 32] {
+        storage_prefix(b"MarketCommons", b":__STORAGE_VERSION__:")
+    }
+
+    #[allow(unused)]
+    pub fn get_on_chain_storage_version_of_market_commons_pallet() -> StorageVersion {
+        let key = storage_prefix_of_market_common_pallet();
+        unhashed::get_or_default(&key)
+    }
+
+    #[allow(unused)]
+    pub fn put_storage_version_of_market_commons_pallet(value: u16) {
+        let key = storage_prefix_of_market_common_pallet();
+        unhashed::put(&key, &StorageVersion::new(value));
+    }
+
+    #[allow(unused)]
+    const SWAPS: &[u8] = b"Swaps";
+    #[allow(unused)]
+    const POOLS: &[u8] = b"Pools";
+    #[allow(unused)]
+    fn storage_prefix_of_swaps_pallet() -> [u8; 32] {
+        storage_prefix(b"Swaps", b":__STORAGE_VERSION__:")
+    }
+    #[allow(unused)]
+    pub fn key_to_hash<H, K>(key: K) -> Vec<u8>
+    where
+        H: StorageHasher,
+        K: Encode,
+    {
+        key.using_encoded(H::hash).as_ref().to_vec()
+    }
+    #[allow(unused)]
+    pub fn get_on_chain_storage_version_of_swaps_pallet() -> StorageVersion {
+        let key = storage_prefix_of_swaps_pallet();
+        unhashed::get_or_default(&key)
+    }
+    #[allow(unused)]
+    pub fn put_storage_version_of_swaps_pallet(value: u16) {
+        let key = storage_prefix_of_swaps_pallet();
+        unhashed::put(&key, &StorageVersion::new(value));
+    }
+}
```

Comment:
What do we need this for? I can't find where this is used.

```suggestion
```

Reply:
If we keep it around it may be just hand for next time when we write migration. However I can remove it if you prefer that.

Reply:
I think it's okay to leave it in, because of the common use. That's why we used a lot of `#[allow(unused)]`.

Reply:
We have exactly the same code in the prediction markets pallet. We should deduplicate it.